v2/account_claims.go

```diff
@@ -72,26 +71,36 @@ func (j *JetStreamLimits) IsJSEnabled() bool {
 	return j.MemoryStorage != 0 || j.DiskStorage != 0
 }
 
+type TieredLimits map[string]JetStreamLimits
+
 // OperatorLimits are used to limit access by an account
 type OperatorLimits struct {
 	NatsLimits
 	AccountLimits
 	JetStreamLimits
+	TieredLimits `json:"tiered_limits,omitempty"`
 }
 
-// IsEmpty returns true if all of the limits are 0/false.
+// IsEmpty returns true if all limits are 0/false/empty.
 func (o *OperatorLimits) IsEmpty() bool {
-	return *o == OperatorLimits{}
+	return o.NatsLimits == NatsLimits{} &&
+		o.AccountLimits == AccountLimits{} &&
+		o.JetStreamLimits == JetStreamLimits{} &&
+		len(o.TieredLimits) == 0
 }
 
 // IsUnlimited returns true if all limits are unlimited
 func (o *OperatorLimits) IsUnlimited() bool {
-	return o.AccountLimits.IsUnlimited() && o.NatsLimits.IsUnlimited() && o.JetStreamLimits.IsUnlimited()
+	return o.AccountLimits.IsUnlimited() && o.NatsLimits.IsUnlimited() &&
+		o.JetStreamLimits.IsUnlimited() && len(o.TieredLimits) == 0
 }
 
 // Validate checks that the operator limits contain valid values
-func (o *OperatorLimits) Validate(_ *ValidationResults) {
+func (o *OperatorLimits) Validate(vr *ValidationResults) {
 	// negative values mean unlimited, so all numbers are valid
+	if len(o.TieredLimits) > 0 && o.JetStreamLimits.IsJSEnabled() {
+		vr.AddError("JetStream Limits and tiered JetStream Limits are mutually exclusive")
```

Comment:
So this is strange, but makes a point of TieredLimits not being inside JetStreamLimits. If TieredLimits are still JS limits, it is strange that they are mutually exclusive. Is that simply to say that if Tiered were inside JetStreamLimits, it would mean that all the other configuration inside JetStreamLimits would have to be "empty" if Tiered would not be?

Reply:
basically they are mutually exclusive. if you have tiered limits that contain JetStream Limits, then you can't. have non Tiered JetStreamLimits

Reply:
Ok